packages/ui/src/components/Organisms/PageContent/BlockTeaserList.tsx

```diff
@@ -0,0 +1,88 @@
+import { useIntl } from '@amazeelabs/react-intl';
+import {
+  BlockTeaserListFragment,
+  Locale,
+  TeaserListQuery,
+} from '@custom/schema';
+import queryString from 'query-string';
+import React from 'react';
+
+import { useOperation } from '../../../utils/operation';
+import { CardItem } from '../Card';
+
+export type TeaserListQueryArgs = {
+  title: string | undefined;
+  pageSize: string | undefined;
+  excludeIds: string | undefined;
+};
+
+function getUUIDFromId(id: string) {
+  const words = id.split(':');
+  if (words.length === 1) {
+    return id;
+  }
+  return words[1];
+}
+
+export function BlockTeaserList(props: BlockTeaserListFragment) {
+  const staticIds: Array<string | undefined> = [];
+  return (
+    <div className="bg-white py-12 px-6 lg:px-8">
+      <div className="mx-auto max-w-6xl">
+        <ul className="my-8 grid md:grid-cols-2 lg:grid-cols-3 gap-4 lg:gap-8">
+          {props.staticContent?.map((teaserItem) => {
+            staticIds.push(getUUIDFromId(teaserItem?.content?.id || ''));
```

Comment:
This feels a bit strange here, instead of before the `return` block, although it would mean calling `map` twice (once before `return` and once during the `return`), it would be computed before React tries to render and return any HTML so the `DynamicTeaserList` should only get rendered once and not updated on each `staticIds.push` event